app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorPublishedView.java

```diff
@@ -0,0 +1,507 @@
+    Bundle favorBundle = new Bundle();
+    favorBundle.putParcelable(CommonTools.FAVOR_VALUE_ARGS, newFavor);
+    favorBundle.putString(
+        CommonTools.FAVOR_SOURCE, getString(R.string.favor_source_publishedFavor));
+    findNavController(requireActivity(), R.id.nav_host_fragment)
+        .navigate(R.id.action_global_favorEditingView, favorBundle);
+  }
+
+  private void commitFavor() {
+    currentFavor.setAccepterId(currentUser.getUid());
+    CompletableFuture updateFuture = getViewModel().commitFavor(currentFavor, 1);
+    updateFuture.thenAccept(
+        o ->
+            CommonTools.showSnackbar(requireView(), getString(R.string.favor_respond_success_msg)));
+    updateFuture.exceptionally(onFailedResult(requireView()));
+  }
+
+  private void cancelCommit() {
+    for (int i = 1; i < currentFavor.getUserIds().size(); i++)
+      if (currentFavor.getUserIds().get(i).equals(currentUser.getUid()))
+        currentFavor.getUserIds().remove(i);
+    CompletableFuture updateFuture = getViewModel().commitFavor(currentFavor, -1);
+    updateFuture.thenAccept(
+        o ->
+            CommonTools.showSnackbar(requireView(), getString(R.string.favor_respond_success_msg)));
+    updateFuture.exceptionally(onFailedResult(requireView()));
+  }
+
+  private void acceptFavor(User user) {
+    CompletableFuture acceptFavorFuture = getViewModel().acceptFavor(currentFavor, user);
+    acceptFavorFuture.thenAccept(
+        o -> {
+          CommonTools.showSnackbar(getView(), getString(R.string.favor_respond_success_msg));
+          UserUtil.getSingleInstance()
+              .findUser(user.getId())
+              .thenAccept(
+                  user1 -> {
+                    user1.setAcceptedFavors(user1.getAcceptedFavors() + 1);
+                    UserUtil.getSingleInstance().updateUser(user1);
+                  });
+        });
+    // acceptFavorFuture.exceptionally(handleException());
+  }
+
+  private void completeFavor() {
+    CompletableFuture updateFuture = getViewModel().completeFavor(currentFavor, isRequested);
+    updateFuture.thenAccept(
+        o ->
+            CommonTools.showSnackbar(
+                requireView(), getString(R.string.favor_complete_success_msg)));
+    updateFuture.exceptionally(onFailedResult(requireView()));
+  }
+
+  private void cancelFavor() {
+    // remove the potential helpers before cancel
+    if (favorStatus == FavorStatus.REQUESTED)
+      for (int i = 1; i < currentFavor.getUserIds().size(); i++)
+        currentFavor.getUserIds().remove(i);
+    CompletableFuture completableFuture = getViewModel().cancelFavor(currentFavor, isRequested);
+    completableFuture.thenAccept(
+        o -> CommonTools.showSnackbar(getView(), getString(R.string.favor_cancel_success_msg)));
+    completableFuture.exceptionally(onFailedResult(requireView()));
+  }
+
+  private void deleteFavor() {
+    CompletableFuture deleteFuture = getViewModel().deleteFavor(currentFavor);
```

Comment:
Should only be deletable if it's requested and if nobody has accepted the favor before. I think right now even the accepter can delete the favor right?

Reply:
oh, I misunderstand before. fixed now.